testing/scenario_app/ios/Scenarios/ScenariosUITests/iPadGestureTests.m

```diff
@@ -62,23 +69,123 @@ - (void)testPointerButtons {
 
   [flutterView tap];
   // Initial add event should have buttons = 0
-  XCTAssertTrue([app.textFields[@"PointerChange.add:0"] waitForExistenceWithTimeout:1],
-                @"PointerChange.add event did not occur");
+  XCTAssertTrue(
+      [app.textFields[@"0,PointerChange.add,device=0,buttons=0"] waitForExistenceWithTimeout:1],
+      @"PointerChange.add event did not occur for a normal tap");
   // Normal tap should have buttons = 0, the flutter framework will ensure it has buttons = 1
-  XCTAssertTrue([app.textFields[@"PointerChange.down:0"] waitForExistenceWithTimeout:1],
-                @"PointerChange.down event did not occur for a normal tap");
-  XCTAssertTrue([app.textFields[@"PointerChange.up:0"] waitForExistenceWithTimeout:1],
-                @"PointerChange.up event did not occur for a normal tap");
+  XCTAssertTrue(
+      [app.textFields[@"1,PointerChange.down,device=0,buttons=0"] waitForExistenceWithTimeout:1],
+      @"PointerChange.down event did not occur for a normal tap");
+  XCTAssertTrue(
+      [app.textFields[@"2,PointerChange.up,device=0,buttons=0"] waitForExistenceWithTimeout:1],
+      @"PointerChange.up event did not occur for a normal tap");
   SEL rightClick = @selector(rightClick);
   XCTAssertTrue([flutterView respondsToSelector:rightClick],
                 @"If supportsPointerInteraction is true, this should be true too.");
   [flutterView performSelector:rightClick];
-  // Since each touch is its own device, we can't distinguish the other add event(s)
+  // On simulated right click, a hover also occurs, so the hover pointer is added
+  XCTAssertTrue(
+      [app.textFields[@"3,PointerChange.add,device=1,buttons=0"] waitForExistenceWithTimeout:1],
+      @"PointerChange.add event did not occur for a right-click's hover pointer");
+  [NSThread sleepForTimeInterval:5.0];  // The hover pointer is removed after ~3.5 seconds, this
+                                        // ensures all events have been received
+  // The hover events are interspersed with the right-click events in a varying order
+  // Ensure the individual orderings are respected without hardcoding the absolute sequence
+  NSMutableDictionary* messages = [NSMutableDictionary dictionary];
+  for (XCUIElement* element in [app.textFields allElementsBoundByIndex]) {
+    NSString* rawMessage = element.value;
+    NSUInteger commaIndex = [rawMessage rangeOfString:@","].location;
```

Comment:
This section is hard to understand, what is it doing, exactly?  Does `-componentsSeparatedByString` make it easier to read?

Reply:
I just need the first component, I don't want to split the whole string by commas since there are more commas later.